sklearn/ensemble/tests/test_iforest.py

```diff
@@ -356,3 +356,23 @@ def test_iforest_preserve_feature_names():
     with warnings.catch_warnings():
         warnings.simplefilter("error", UserWarning)
         model.fit(X)
+
+
+@pytest.mark.parametrize("sparse_container", CSC_CONTAINERS + CSR_CONTAINERS)
+def test_iforest_sparse_input_float_contamination(sparse_container):
+    """Check that `IsolationForest` accepts sparse matrix input and float value for
+    contamination.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/27626
+    """
+    X, _ = make_classification(n_samples=50, n_features=4, random_state=0)
+    X = sparse_container(X)
+    X.sort_indices()
+    contamination = 0.1
+    iforest = IsolationForest(
+        n_estimators=5, contamination=contamination, random_state=0
+    ).fit(X)
+
+    X_decision = iforest.decision_function(X)
+    assert (X_decision < 0).sum() / X.shape[0] == pytest.approx(contamination)
```

Comment:
Would it be possible to use math.isclose here?

Reply:
This is perfectly fine especially if it is the standard way. I was just thinking along the lines that we usually use the close methods found in numpy for asserting so maybe we could do something similar here.

Reply:
Ah I see. I think that in the test itself, we (at least I :)) tend to use `pytest.approx`. I keep `math.isclose` or `np.isclose` for within the non-test code source.